tests/unit/LSYiiValidatorsTest.php

```diff
@@ -27,4 +27,155 @@ public function testHtmlEntityDecodeFilter()
+        $this->assertFalse($superAdminValidator->xssfilter, 'The xssfilter attribute should be false for super admins.');
+
+        //Changing filterxsshtml.
+        \Yii::app()->setConfig('filterxsshtml', false);
+        $newSuperAdminValidator = new \LSYii_Validators();
+
+        $this->assertFalse(\Yii::app()->getConfig('filterxsshtml'), 'filterxsshtml was just changed to false.');
+        $this->assertFalse($newSuperAdminValidator->xssfilter, 'The xssfilter attribute should be false for super admins.');
+
+        //Returning to original values.
+        \Yii::app()->setConfig('filterxsshtml', true);
+        \Yii::app()->session['loginID'] = null;
+    }
+
+    /**
+     * Testing that the xssfilter attribute varies
+     * for regular users depending on filterxsshtml.
+     */
+    public function testXssFilterAttributeForRegularUsers()
+    {
+        //Create user.
+        $newPassword = createPassword();
+        $userId = \User::insertUser('test_user', $newPassword, 'John Doe', 1, 'jd@mail.com');
+
+        //Mocking regular user login.
+        \Yii::app()->session['loginID'] = $userId;
+        $regularUserValidator = new \LSYii_Validators();
+
+        $this->assertTrue(\Yii::app()->getConfig('filterxsshtml'), 'filterxsshtml should be true by default.');
+        $this->assertTrue($regularUserValidator->xssfilter, 'The xssfilter attribute should be true for regular users.');
+
+        //Changing filterxsshtml.
+        \Yii::app()->setConfig('filterxsshtml', false);
+        $newRegularUserValidator = new \LSYii_Validators();
+
+        $this->assertFalse(\Yii::app()->getConfig('filterxsshtml'), 'filterxsshtml was just changed to false.');
+        $this->assertFalse($newRegularUserValidator->xssfilter, 'The xssfilter attribute should be false for regular users with filterxsshtml set to false.');
+
+        //Returning to original values.
+        \Yii::app()->setConfig('filterxsshtml', true);
+        \Yii::app()->session['loginID'] = null;
+
+        //Delete user.
+        $user = \User::model()->findByPk($userId);
+        $user->delete();
+    }
+
+    /**
+     * Testing that any script or dangerous HTML is removed.
+     */
+    public function testXssFilterApplied()
+    {
+        $validator = new \LSYii_Validators();
+
+        $cases = array(
+            array(
+                'string'   => '<script>alert(`Test`)</script>',
+                'expected' => ''
+            ),
+            array(
+                'string'   => `{join('html_entity_decode("', '<script>alert("Test")</script>")')}`,
+                'expected' => ''
+            ),
+            array(
+                'string'   => '<title>html_entity_decode("<script>alert("Test")</script>")</title>',
+                'expected' => 'html_entity_decode("")'
+            ),
+            array(
+                'string'   => `{join('html_entity_decode("', '<s', 'cript>alert("Test")</script>")')}`,
```

Comment:
Must test too `{join("<s", "cript>alert("Test")</script>")}` see https://github.com/LimeSurvey/LimeSurvey/blob/a3605854a63d56b360660dea47e144ea766d382f/application/core/LSYii_Validators.php#L146 no ?

Alls DS_STRING and DQ_STRING are purified

Reply:
Yes, will add it.
We just grab the ones from the ticket.

I guess even more can be added.
The idea is to add more as they appear

Reply:
Yes, but here : you check more than the previous pone.

if `{join("<s", "cript>alert("Test")</script>")}` is available, no need `html_entity_decode` ;)

Reply:
The expression for testing :` {join("<","script,">",'alert("Test")',"<","/script",">")}`

Must return `&lt;script&gt;alert("Test")&lt;/script&gt;`

Reply:
Added it below.